src/Access/ContextAccess.cpp

```diff
@@ -176,12 +176,12 @@ void ContextAccess::setRolesInfo(const std::shared_ptr<const EnabledRolesInfo> &
 }
 
 
-bool ContextAccess::isCorrectPassword(const String & password) const
+bool ContextAccess::isCorrectPassword(const String & password, const ExternalAuthenticators & external_authenticators) const
```

Comment:
It seems it's better to make `std::shared_ptr<const ExternalAuthenticators>` a part of the class `ContextAccess`. Doing that way would allow to simplify the interface because you would have to pass only password to `ContextAccess::isCorrectPassword()`.

Reply:
Technically, it will require significantly more copies/ptrs passing around, since `ContextAccess` instances are stored in
```
Poco::ExpireCache<ContextAccess::Params, std::shared_ptr<const ContextAccess>> cache;
```
...which is a member of `ContextAccessCache`, an instance of which is stored in `AccessControlManager`. The `ExternalAuthenticators` is constructed when `AccessControlManager` parses config, and it should be shared across all instances of `ContextAccess`. So that will be really messy.

Reply:
I have a better idea: `ContextAccess` already have a pointer to `AccessControlManager`, so it can get a reference to `ExternalAuthenticators` anytime so you need neither to pass `ExternalAuthenticators` to `ContextAccess::isCorrectPassword()` nor to share pointers to `ExternalAuthenticators` in `ContextAccess`.

Reply:
Yeah, that should work. Pushed.